pkg/storage/endpoint/tso.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpoint
+
+import (
+	"strings"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/utils/typeutil"
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/zap"
+)
+
+// TSOStorage defines the storage operations on the TSO.
+// global tso: key is `/microservice/tso/{keyspaceGroupName}/timestamp`
+// local tso: key is `/microservice/tso/{keyspaceGroupName}/lts/{dcLocation}/timestamp`
+// FIXME: When we upgrade from the old version, there may be compatibility issues.
+type TSOStorage interface {
+	LoadTimestamp(keyspaceGroupName string, dcLocationKey ...string) (time.Time, error)
+	SaveTimestamp(keyspaceGroupName string, ts time.Time, dcLocationKey ...string) error
+}
+
+var _ TSOStorage = (*StorageEndpoint)(nil)
+
+// LoadTimestamp loads a timestamp from the storage according to the keyspaceGroupName and dcLocation.
+func (se *StorageEndpoint) LoadTimestamp(keyspaceGroupName string, dcLocationKey ...string) (time.Time, error) {
+	var prefix string
+	if len(dcLocationKey) != 0 {
+		prefix = timestampPrefix(keyspaceGroupName) + dcLocationKey[0] + "/"
```

Comment:
Can we use a string buffer to concatenate the `prefix` rather than the operator `+`? Since it's on the TSO path, I think we should be efficient as possible.

Reply:
sounds good